src/builder.rs

```diff
@@ -206,14 +217,30 @@ pub enum InitError<DI, P> {
     /// This error is returned when the configuration passed to the builder is
     /// invalid. For example, when the combination of bit depth and interface
     /// kind isn't supported by the selected model.
-    InvalidConfiguration,
+    InvalidConfiguration(ConfigurationError),
+}
+
+/// Specifics of [InitError::InvalidConfiguration] if configuration was found invalid
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum ConfigurationError {
+    /// Display Model and Interface kind are mismatching
+    ModelInterfaceMismatch,
+    /// Display dimensions provided in [Builder::display_size] were invalid, e.g. width or height of 0
+    DisplayDimensions,
```

Comment:
I think we should keep the naming consistent and use "display size" and not "display dimensions".

```suggestion
    /// Invalid display size.
    /// 
    /// At least one dimension of the [`display_size`](Builder::display_size) is
    /// invalid. The display size must be larger than zero and less or equal to
    /// the [`FRAMEBUFFER_SIZE`](Model::FRAMEBUFFER_SIZE) for the selected
    /// model.
    InvalidSize,   
```

Reply:
Why didn't you use this doc comment, but the other two?

Reply:
fixed